### PR TITLE
Create TypeScript type definition

### PR DESCRIPTION
**Pull request by KSXGitHub**
So that I can use this package in my TypeScript project

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  interface PNPM {
+    readonly name: 'pnpm'
+    readonly version: string
+  }
+
+  interface ScopedPNPM {
+    readonly name: '@zkochan/pnpm'
+    readonly version: string
```
Comment (zkochan) [inline index.d.ts]: I think it should be something more generic. like {name: string} | {name:string, version: string}

Comment (KSXGitHub) [reply index.d.ts]: How about this?

```typescript
interface ScopedPNPM {
  readonly name: string
  readonly version?: string
}
```

Comment (zkochan) [reply index.d.ts]: ok, but maybe call it just OtherPM

Comment (KSXGitHub) [reply index.d.ts]: It will be done.

**Comment by KSXGitHub** [comment]
@zkochan I committed additional change to fix my mistake of not update all references. Please merge it again!

**Comment by zkochan** [comment]
ok

**Pull request merged**